src/batch/src/executor/generic_exchange.rs

```diff
@@ -164,26 +167,33 @@ impl<CS: 'static + Send + CreateSource, C: BatchTaskContext> Executor
 impl<CS: 'static + Send + CreateSource, C: BatchTaskContext> GenericExchangeExecutor<CS, C> {
     #[try_stream(boxed, ok = DataChunk, error = BatchError)]
     async fn do_execute(self: Box<Self>) {
-        let mut stream = select_all(
-            self.proto_sources
-                .into_iter()
-                .zip_eq_fast(self.source_creators)
-                .map(|(prost_source, source_creator)| {
-                    Self::data_chunk_stream(
-                        prost_source,
-                        source_creator,
-                        self.context.clone(),
-                        self.metrics.clone(),
-                        self.identity.clone(),
-                    )
-                })
-                .collect_vec(),
-        )
-        .boxed();
-
-        while let Some(data_chunk) = stream.next().await {
-            let data_chunk = data_chunk?;
-            yield data_chunk
+        let streams = self
```

Comment:
How about move this part to the `self.sequential` block? It's not easy to understand the difference of collection.